backend/rest_api_service/main.py

```diff
@@ -28,6 +29,81 @@ async def get_all_users(response: Response, db: Session = Depends(get_db)):
         return {"message": f"An error occured: {e}"}
 
 
+
+
+
+@app.get("/users/get_user_by_id/{id}", tags = ['User'], status_code=status.HTTP_200_OK)
+async def get__user_by_id(id : int, response: Response, db: Session = Depends(get_db)):
+    try:
+        user = await crud.get_user_by_id(db, id)
+        print(id)
+        return user
+
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occured: {e}"}
+
+@app.get("/users/get_user_by_username/{username}", tags = ['User'], status_code=status.HTTP_200_OK)
+async def get__user_by_username(username : str, response: Response, db: Session = Depends(get_db)):
+    try:
+        user = await crud.get_user_by_username(db, username)
+        print(username)
+        return user
+
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occured: {e}"}
+    
+@app.get("/users/get_all_events_by_user_id/{user_id}", tags=['User'], status_code=status.HTTP_200_OK)
+async def get_all_events_by_user_id(user_id: int, response: Response, db: Session = Depends(get_db)):
+    try:
+        user = await crud.get_user_by_id(db, user_id)
+        if user:
+            events_created = user.events_created
+            return events_created
+        else:
+            response.status_code = status.HTTP_404_NOT_FOUND
+            return {"message": "User not found"}
+    except Exception as e:
+        response.status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
+        return {"message": f"An error occurred: {e}"}
+
+@app.get("/users/get_all_friends_by_user_id/{user_id}", tags=['User'], status_code=status.HTTP_200_OK)
+async def get_all_friends_by_user_id(user_id: int, response: Response, db: Session = Depends(get_db)):
+    try:
+        user_friends = await crud.get_all_friends_by_user_id(db, user_id)
+        if user_friends:
+            return user_friends
+        response.status_code = status.HTTP_404_NOT_FOUND
+        return {"message": "User not found or user has no friends"}
+    except Exception as e:
+        response.status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
+        return {"message": f"An error occurred: {e}"}
+
+@app.post("/users/create_user", tags=['User'], status_code=status.HTTP_201_CREATED)
```

Comment:
to już nie będzie potrzebne, bo zostało juz zaimplementowane przez Kacpra i jest w mainie w endpoincie `register`